tests/gis_tests/test_gis_tests_utils.py

```diff
@@ -0,0 +1,52 @@
+from django.db import connection, models
+from django.db.models.expressions import Func
+from django.test import SimpleTestCase
+
+from .utils import FuncTestMixin
+
+
+def test_mutation(raises=True):
+    def wrapper(mutation_func):
+        def test(test_case_instance, *args, **kwargs):
+            class TestFunc(Func):
+                output_field = models.IntegerField()
+
+                def __init__(self):
+                    self.attribute = 'initial'
+                    super().__init__('initial', ['initial'])
+
+                def as_sql(self, *args, **kwargs):
+                    mutation_func(self)
+                    return '', ()
+
+            if raises:
+                msg = 'TestFunc Func was mutated during compilation.'
```

Comment:
I'm not sure if message should be tested more precisely.

Reply:
Looks okay.